.github/workflows/build_and_test_cosmian_vm.yml

```
@@ -0,0 +1,174 @@
+---
+name: Build and Publish VM Image
+
+on:
+  workflow_call:
+    inputs:
+      distrib:
+        required: true
+        type: string
+
+env:
+  GCP_DEV_PROJECT: cosmian-dev
+  GCP_PUBLIC_PROJECT: cosmian-public
+
+jobs:
+  build-kms-cosmian-vm-image:
+    name: Build KMS in Cosmian VM Image for SEV
+    runs-on: ubuntu-22.04
+    outputs:
+      timestamp: ${{ steps.env.outputs.TIMESTAMP }}
+      image_name: ${{ steps.env.outputs.IMAGE_NAME }}
+      ci_instance: ${{ steps.env.outputs.CI_INSTANCE }}
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    defaults:
+      run:
+        working-directory: ./packer
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Create env variables
+        id: env
+        run: |
+          TIMESTAMP="$(date -u +'%Y%m%d%H%M%S')"
+          echo "TIMESTAMP=$TIMESTAMP" >> "$GITHUB_OUTPUT"
+          echo "IMAGE_NAME=alpha-$TIMESTAMP-cosmian-vm-kms-sev-${{ inputs.distrib }}" >> "$GITHUB_OUTPUT"
+          echo "CI_INSTANCE=gh-ci-cosmian-vm-kms-$TIMESTAMP-${{ inputs.distrib }}" >> "$GITHUB_OUTPUT"
+
+      - name: Setup `packer`
+        uses: hashicorp/setup-packer@main
+
+      - name: Run `packer init`
+        run: packer init -machine-readable gcp-sev-${{ inputs.distrib }}.pkr.hcl
+
+      - name: Install plugins
+        run: |
+          packer plugins install github.com/hashicorp/amazon
+          packer plugins install github.com/hashicorp/googlecompute
+
+      # - name: Authenticate to AWS project
+      #   uses: aws-actions/configure-aws-credentials@v1
+      #   with:
+      #     aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      #     aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      #     aws-region: eu-west-1
+
+      - name: Authenticate to Google Cloud project
+        uses: "google-github-actions/auth@v2"
+        with:
+          credentials_json: "${{ secrets.GOOGLE_COSMIAN_DEV_CREDENTIALS }}"
+
+      - name: Build GCP images - main
+        env:
+          TIMESTAMP: ${{ steps.env.outputs.TIMESTAMP }}
+        if: ${{ !startsWith(github.ref, 'refs/tags') }}
```

could this be working?
```suggestion
        if: !startsWith(github.ref, 'refs/tags/')
```